galleon-plugins/src/main/java/org/wildfly/galleon/plugin/Utils.java

```diff
@@ -130,11 +135,61 @@ public static MavenArtifact toArtifactCoords(Map<String, String> versionProps, S
             if (parts.length < 3) {
                 throw new ProvisioningException("Failed to resolve the version for artifact: " + resolvedStr);
             }
-            artifact.setVersion(parts[2]);
+            String version = resolveExpression(str, parts[2]);
+            artifact.setVersion(version);
         }
         return artifact;
     }
 
+    /**
+     * Resolve an expression composed of ${a,b,c;defaultValue} Where a, b and c can be System properties or env.XXX env variables.
+     */
+    private static String resolveExpression(String coords, String str) throws ProvisioningException {
+        if (str == null) {
+            return str;
+        }
+        String resolved = str;
+        str = str.trim();
+        if (str.startsWith(EXPRESSION_PREFIX) && str.endsWith(EXPRESSION_SUFFIX)) {
+            String expressions = str.substring(EXPRESSION_PREFIX.length(), str.length() - EXPRESSION_SUFFIX.length());
+            int defValueSeparator = expressions.indexOf(EXPRESSION_DEFAULT_VALUE_SEPARATOR);
+            String defaultValue = null;
+            if (defValueSeparator >= 0) {
+                defaultValue = expressions.substring(defValueSeparator+1, expressions.length());
+                defaultValue = defaultValue.trim();
+                expressions = expressions.substring(0, defValueSeparator);
+            }
+            String[] split = expressions.split(",", -1);
+            String value;
+            for (String expression : split) {
+                expression = expression.trim();
+                if (expression.isEmpty()) {
+                   throw new ProvisioningException("Invalid syntax for expression " + coords);
+                }
+                if (expression.startsWith(EXPRESSION_ENV_VAR)) {
```

Comment:
@jmesnil Do you want to require the 'env.' prefix?  Doing that is consistent with other use cases, but otoh we've talked about moving past that.

It probably doesn't much matter here though; the use case for this is very small and it doesn't hurt us in our expected uses to include the env. in the feature pack.

Reply:
I'd prefer we keep the `env.` prefix for consistency sake. This aligns with general use of expressions in WildFly and don't require the user to think about who will read that expression (the feature-pack or the provisioned WildFly runtime).

Reply:
I kept env.